agent/01_agent_requirements.sh

```diff
@@ -6,10 +6,43 @@ SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/.." && pwd )"
 
 source $SCRIPTDIR/validation.sh
 source $SCRIPTDIR/common.sh
+source $SCRIPTDIR/agent/common.sh
 
 early_deploy_validation
 
 if [[ -z ${AGENT_E2E_TEST_SCENARIO} ]]; then
     echo "AGENT_E2E_TEST_SCENARIO is missing or empty. Did you forget to set the AGENT_E2E_TEST_SCENARIO env var in the config_<USER>.sh file? Supported values: COMPACT_IPV4, COMPACT_IPV6, HA_IPV4, HA_IPV6, SNO_IPV4, SNO_IPV6."
     exit 1
 fi
+
+if [[ ! -z "${OC_MIRROR}" ]] && [[ "${OC_MIRROR}" == true ]]; then
+
+   mkdir -p ${WORKING_DIR}/mirror-registry
+   pushd ${WORKING_DIR}/mirror-registry
+   # run the exec in this dir as execution-environment.tar is also needed
+   mirror_registry_file=mirror-registry.tar.gz
+   mirror_registry_exec=${mirror_registry_file%%.*}
+   if [[ ! -f "./${mirror_registry_exec}" ]]; then
+      curl -O -L https://developers.redhat.com/content-gateway/rest/mirror/pub/openshift-v4/clients/mirror-registry/latest/${mirror_registry_file}
+      tar xzf ${mirror_registry_file}
+      chmod +x ${mirror_registry_exec}
+      rm -f ${mirror_registry_file}
+   fi
+   popd
+
+   oc_mirror_file=oc-mirror.tar.gz
+   oc_mirror_exec=${oc_mirror_file%%.*}
+   if [[ ! -f "/usr/local/bin/${oc_mirror_exec}" ]]; then
+      curl -O -L https://mirror.openshift.com/pub/openshift-v4/x86_64/clients/ocp/stable/${oc_mirror_file}
+      tar xzf ${oc_mirror_file}
+      chmod +x ${oc_mirror_exec}
+      sudo mv -f ${oc_mirror_exec} /usr/local/bin
+      rm -f ${oc_mirror_file}
+   fi
+
+   # set up the channel using the most recent candidate release
+   release_candidate=`oc mirror list releases --channel=candidate-${OPENSHIFT_RELEASE_STREAM} | tail -1`
```

Comment:
As discussed, it looks like this one is ignoring `OPENSHIFT_RELEASE_IMAGE` when set, thus resulting in a different version installed.
Additionally, if possible it would be preferrable to set `OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE` in one of the existing spots to minimize the points where it gets touched.
Even better, but very likely to be addressed in another PR, to avoid using that var for disconnected setups

Reply:
> As discussed, it looks like this one is ignoring `OPENSHIFT_RELEASE_IMAGE` when set, thus resulting in a different version installed. Additionally, if possible it would be preferrable to set `OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE` in one of the existing spots to minimize the points where it gets touched. Even better, but very likely to be addressed in another PR, to avoid using that var for disconnected setups

The reason it's using `4.11.6` when `OPENSHIFT_RELEASE_STREAM=4.12` is because for oc-mirror the ImageSetConfiguration requires a "candidate" channel and there are currently no "candidate" releases for 4.12, the most recent candidate release is `4.11.6`

Reply:
Ok, how can then be used `oc mirror` with a custom provided version, since that's the case we'll have also in the CI?